statsrelay.go

```diff
@@ -88,12 +94,33 @@ func getSockBufferMaxSize() (int, error) {
 // the metric key name and returns it as a string.
 func getMetricName(metric []byte) (string, error) {
 	// statsd metrics are of the form:
-	//    KEY:VALUE|TYPE|RATE
+	//    KEY:VALUE|TYPE|RATE or KEY:VALUE|TYPE|RATE|#tags
+	var metricName string
 	length := bytes.IndexByte(metric, byte(':'))
 	if length == -1 {
 		return "error", errors.New("Length of -1, must be invalid StatsD data")
 	}
-	return string(metric[:length]), nil
+	if len(metricsPrefix) != 0 {
+		return genPrefix(metric[:length], metricsPrefix), nil
+	} else {
```

Comment:
It would be more idiomatic code here to drop the else clase and simply `return string(metric[:length]), nil`.  This would also no longer require the `metricName` variable.